lib/presets.js

```diff
@@ -428,6 +428,63 @@ module.exports = {
           return callback(error);
         }
       });
+    },
+
+    /**
+     * @summary Update quoted version immediately following a regex
+     * @function
+     * @public
+     *
+     * @param {Object} options - options
+     * @param {String} [options.baseDir] - directory to prefix onto options.file
+     * @param {String} options.file - file to modify
+     * @param {String} options.regex - regex leading up to the quoted version string
+     * @param {String} [options.regexFlags] - any modifier flags to apply to the regex
+     * @param {String} cwd - current working directory
+     * @param {String} version - version
+     * @param {Function} callback - callback (error)
+     * @returns {null}
+     *
+     * @example
+     * presets.updateVersion.quoted({file: "myfile.h", regex: /^VERSION\s+=\s+/}, process.cwd(), '1.0.0', (error) => {
+     *   if (error) {
+     *     throw error;
+     *   }
+     * });
+     */
+    quoted: (options, cwd, version, callback) => {
+      _.defaults(options, {
+        baseDir: '.',
+        regexFlags: ''
+      });
+
+      if (path.isAbsolute(options.baseDir)) {
+        return callback(new Error('baseDir option can\'t be an absolute path'));
+      }
+      if (!options.file) {
+        return callback(new Error('Missing file option'));
+      }
+      if (path.isAbsolute(options.file)) {
+        return callback(new Error('file option can\'t be an absolute path'));
+      }
+      if (!options.regex) {
+        return callback(new Error('Missing regex option'));
+      }
+
+      const updateFile = path.join(cwd, options.baseDir, options.file);
+
+      const cleanedVersion = semver.clean(version);
+
+      if (!cleanedVersion) {
+        return callback(new Error(`Invalid version: ${version}`));
+      }
+
+      replace(
+        updateFile,
+        new RegExp('(' + RegExp(options.regex).source + ')("|\').*?\\2', options.regexFlags),
```

Comment:
What is the reasoning behind piping `options.regex` through `RegExp(...).source`, instead of using it directly for concatenation?

Reply:
If you concatenate it directly, it works for strings but not for regex literals
```js
$ node
> '(' + 'version\\s*=\\s*' + ')("|\').*?\\2'
'(version\\s*=\\s*)("|\').*?\\2'
> '(' + /version\s*=\s*/ + ')("|\').*?\\2'
'(/version\\s*=\\s*/)("|\').*?\\2'
> 
> '(' + RegExp('version\\s*=\\s*').source + ')("|\').*?\\2'
'(version\\s*=\\s*)("|\').*?\\2'
> '(' + RegExp(/version\s*=\s*/).source + ')("|\').*?\\2'
'(version\\s*=\\s*)("|\').*?\\2'
```
See also https://stackoverflow.com/a/185529

(although this is the first time I've used regexes like this in javascript, so there might well be a better way of doing this!)